lib/ansible/parsing/plugin_docs.py

```diff
@@ -56,10 +56,19 @@ def read_docstring(filename, verbose=True, ignore_errors=True):
                         if isinstance(child.value, ast.Dict):
                             data[varkey] = ast.literal_eval(child.value)
                         else:
-                            if theid in ['EXAMPLES', 'RETURN']:
-                                # examples 'can' be yaml, return must be, but even if so, we dont want to parse as such here
+                            if theid == 'EXAMPLES':
+                                # examples 'can' be yaml, but even if so, we dont want to parse as such here
```

Comment:
In ansible-base we've required examples to be valid yaml for quite some time:

https://github.com/ansible/ansible/blob/840d3a9dd7d1233ab55aa9516fc9b44e4c865a81/test/lib/ansible_test/_data/sanity/yamllint/yamllinter.py#L171-L175

Reply:
I guess the main concern are third-party plugins that weren't tested with `ansible-test`. OTOH this only affects `ansible-doc` and `ansible-console`'s help, so I think it is an acceptable risk.

Reply:
@mattclay have we made exceptions for certain plugins? some are using ini files in examples ?

Reply:
Not that I'm aware of.

Reply:
```
EXAMPLES = r'''
    # define 2 hosts in command line
    # ansible -i '10.10.2.6, 10.10.2.4' -m ping all

    # DNS resolvable names
    # ansible -i 'host1.example.com, host2' -m user -a 'name=me state=absent' all

    # just use localhost
    # ansible-playbook -i 'localhost,' play.yml -c local
```
seems people have been 'creative' to avoid the restriction, we really should not make this a strict requirement

Reply:
specially since nothing will display if we treat it as yaml (currently its processed as string)

Reply:
```
EXAMPLES = '''
  example1: |
      # example cfg file
      [web]
      host1
      host2 ansible_port=222 # defined inline, interpreted as an integer

      [web:vars]
```

Reply:
I know that examples is not valid structured yaml for a large number of plugins.  Perhaps yamllint decides it's a valid yaml string?